src/kubernetes_charm.py

```diff
@@ -128,7 +128,10 @@ def _patch_service(self) -> None:
     # =======================
 
     def _on_install(self, _) -> None:
-        """Patch existing k8s service to include read-write and read-only services."""
+        """Open ports & patch k8s service."""
+        if ops.JujuVersion.from_environ().supports_open_port_on_k8s:
+            for port in (self._READ_WRITE_PORT, self._READ_ONLY_PORT, 6448, 6449):
```

Comment:
Today we are not exposing mysqlx protocol, but IMHO, we should (to avoid rush in the future).
I have added X protocol support to the next sprint roadmap to discuss internally. CC: @delgod